_blog/scaling-dot-net-core-microservices/index.md

```diff
@@ -0,0 +1,403 @@
+
+> TOP TIP: When you build something, learn how to kill it and make it graceful
+
+Regardless of what part of your system this relates to, you should be able to create and delete parts of your system.
+For example, if you're creating a cluster, you should be able to have a command that can delete your cluster.
+If you create a console application, you should manage the `Ctrl + C` or `SIGTERM` event.
+Normally, containers will give containers 30 seconds notice before terminating an application.
+If you have a console application that is processing a lead, how do you manage this when the orchestrator fires the `Ctrl + C` event?
+Do you delete the process or wait for it to complete?  These are the decisions you should be making prior to moving to a container.
+Also, this will empower your team later on, as you'll know how to manage when you have an actual fire.
+
+### Docker Build
+
+We use Docker build to create Docker images.  Docker images can then be used to spin up containers.
+
+> TOP TIP: Environment Variables are your Best Friend Forever (BFF)
+
+When running containers, you have the option to pass variables into the container via a number of ways:
+
+* Volumes
+* Environment Variable
+
+When should you them?  Use volumes for static assets that might need to change.  For example, if you have a website then load your images via a volume, or if you have a database then keep the data within a volume so that you don't lose it when the container is deleted.
+Use Environment Variables for connections strings and any secrets that you're managing with the application.
+
+### Docker Compose
+
+When you have lots of images that need to run together then one option is Docker Compose.  By creating a `docker-compose` file, you can run a command to spin up a local environment within a terminal window and then access the services locally.
+
+## Orchestrator to the rescue
+
+So far, everything we've spoken about has been based on a single computer.  Now I can guess what you're thinking...
+
+> But if all my code is running in production on one box, what happens when that dies?  
+> *Everyone, all the time.*
+
+If you're dependent on any one thing then you're getting ready for a bad time.
+Your application, infrastructure and sanity can only cope with so much.
+Luckily, Orchestrator's like Kubernetes are making this simpler for you to manage.
+Orchestrator's are based on a declarative state, this means you declare of how much you should have and Kubernetes will maintain it.
+So if I want four instances of my website running, I'll get four.  If one instance dies another one will spin up.
+If another instance comes back to life and I have five websites running, then Kubernetes will send a kill command to one of the instances and get it back to four.
+This happens 24/7, without human interaction allowing you to stay asleep and enjoy your life away from a computer.
+
+### Setup the Application
```

Comment:
I think this should be moved further down.
I have a vague idea of what kubernetes is... and you're asking me to use kubectl and YAML.
Perhaps we should introduce minikube first as a way to create a local Kubernetes cluster.